dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -1715,42 +2108,45 @@ bool DynamicDataImpl::set_complex_to_union(DDS::MemberId id, DDS::DynamicData_va
       if (selected_member->get_descriptor(selected_md) != DDS::RETCODE_OK) {
         return false;
       }
-      CORBA::Long disc_val;
-      const DynamicDataImpl* dd_impl = dynamic_cast<const DynamicDataImpl*>(value.in());
-      if (!dd_impl || !dd_impl->read_discriminator(disc_val)) {
+      if (!validate_discriminator(disc_val, selected_md)) {
+        if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+          ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: DynamicDataImpl::set_complex_to_union:"
```

Comment:
NOTICEs should go to ACE_DEBUG.

Reply:
I checked the guidelines and it has ACE_ERROR for NOTICE.

Reply:
Yes, I think we were (or at least I was) kinda uncertain on the dividing line for ACE_ERROR/ACE_DEBUG. I think the rational for guidelines is that the difference only matters for ACE's error status, which we don't use and the warnings and notices are basically less serious errors.

Reply:
I would say we keep NOTICEs with ACE_ERROR if ACE_ERROR and ACE_DEBUG are not much different for NOTICEs. I think we have been using NOTICEs with ACE_ERROR pretty consistently in this file and other files. If we update the guidelines, we can update them all at once.

Reply:
You are correct on this.